.github/workflows/build.yaml

```diff
@@ -31,8 +31,20 @@ jobs:
         strategy:
             matrix:
                 type: [gcc_debug, gcc_release, clang, mbedtls, clang_experimental]
+                logging: [all_logging_enabled]
+                include:
+                    # The gcc_debug job skips the CodeQL checks, so it's faster
```

Comment:
I think it would also be ok to skip `gcc_debug + no_detailed_logging` build. `gcc_debug + all_logging_enabled` and `gcc_release + no_detail_logging` should cover all we need for now.

Reply:
I'm not sure I follow.  The setup here is that we build all types with `all_logging_enabled` and also build just the `gcc_debug` type with `no_detail_logging`.  We're not building `gcc_release` with `no_detail_logging`....

Reply:
Sorry, updated the comment :). But it's just a suggestion given our CI performance issues.

Reply:
Ah, I see.  Yeah, I need to think a bit about whether we can prune some of these bits.... :(